pkg/stanza/operator/parser/keyvalue/keyvalue_test.go

```diff
@@ -572,6 +581,77 @@ key=value`,
 			false,
 			true,
 		},
+		{
+			"custom pair delimiter in quoted value",
+			func(kv *Config) {
+				kv.PairDelimiter = "_"
+			},
+			&entry.Entry{
+				Body: `a=b_c="d_e"`,
+			},
+			&entry.Entry{
+				Attributes: map[string]any{
+					"a": "b",
+					"c": "d_e",
+				},
+				Body: `a=b_c="d_e"`,
+			},
+			false,
+			false,
+		},
+		{
+			"embedded double quotes in single quoted value",
+			func(kv *Config) {},
+			&entry.Entry{
+				Body: `a=b c='this is a "co ol" value'`,
+			},
+			&entry.Entry{
+				Attributes: map[string]any{
+					"a": "b",
+					"c": "this is a \"co ol\" value",
+				},
+				Body: `a=b c='this is a "co ol" value'`,
+			},
+			false,
+			false,
+		},
```

Comment:
I was able to break this test by removing ` value` from the both and expectations. It seems the final `"` is dropped.

```go
		{
			"embedded double quotes in single quoted value",
			func(kv *Config) {},
			&entry.Entry{
				Body: `a=b c='this is a "co ol"'`,
			},
			&entry.Entry{
				Attributes: map[string]any{
					"a": "b",
					"c": "this is a \"co ol\"",
				},
				Body: `a=b c='this is a "co ol"'`,
			},
			false,
			false,
		},
```

Reply:
I think this behavior was present before this change as well due to this bit of code in `parser()`:
```go
key := strings.TrimSpace(strings.Trim(m[0], "\"'"))
value := strings.TrimSpace(strings.Trim(m[1], "\"'"))
```
It was trimming any and all leading or trailing quotes. I fixed this behavior by getting rid of the need to strip the quotes in `parser()` by no longer including them in the strings returned by `splitPairs()`. We'll still call `strings.TrimSpace()` here to strip any leading or trailing white space.

Reply:
Also added in this test as well

Reply:
Nice, thanks